client/driver/docker.go

```diff
@@ -470,63 +440,117 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle
+			Password:      driverConfig.Auth[0].Password,
+			Email:         driverConfig.Auth[0].Email,
+			ServerAddress: driverConfig.Auth[0].ServerAddress,
 		}
+	}
 
-		if authConfigFile := d.config.Read("docker.auth.config"); authConfigFile != "" {
-			if f, err := os.Open(authConfigFile); err == nil {
-				defer f.Close()
-				var authConfigurations *docker.AuthConfigurations
-				if authConfigurations, err = docker.NewAuthConfigurations(f); err != nil {
-					return nil, fmt.Errorf("Failed to create docker auth object: %v", err)
-				}
+	if authConfigFile := d.config.Read("docker.auth.config"); authConfigFile != "" {
+		if f, err := os.Open(authConfigFile); err == nil {
+			defer f.Close()
+			var authConfigurations *docker.AuthConfigurations
+			if authConfigurations, err = docker.NewAuthConfigurations(f); err != nil {
+				return fmt.Errorf("Failed to create docker auth object: %v", err)
+			}
 
-				authConfigurationKey := ""
-				if driverConfig.SSL {
-					authConfigurationKey += "https://"
-				}
+			authConfigurationKey := ""
+			if driverConfig.SSL {
+				authConfigurationKey += "https://"
+			}
 
-				authConfigurationKey += strings.Split(driverConfig.ImageName, "/")[0]
-				if authConfiguration, ok := authConfigurations.Configs[authConfigurationKey]; ok {
-					authOptions = authConfiguration
-				}
-			} else {
-				return nil, fmt.Errorf("Failed to open auth config file: %v, error: %v", authConfigFile, err)
+			authConfigurationKey += strings.Split(driverConfig.ImageName, "/")[0]
+			if authConfiguration, ok := authConfigurations.Configs[authConfigurationKey]; ok {
+				authOptions = authConfiguration
 			}
+		} else {
+			return fmt.Errorf("Failed to open auth config file: %v, error: %v", authConfigFile, err)
 		}
+	}
 
-		err = client.PullImage(pullOptions, authOptions)
-		if err != nil {
-			d.logger.Printf("[ERR] driver.docker: failed pulling container %s:%s: %s", repo, tag, err)
-			return nil, d.recoverablePullError(err, image)
-		}
-		d.logger.Printf("[DEBUG] driver.docker: docker pull %s:%s succeeded", repo, tag)
+	err := client.PullImage(pullOptions, authOptions)
+	if err != nil {
+		d.logger.Printf("[ERR] driver.docker: failed pulling container %s:%s: %s", repo, tag, err)
+		return d.recoverablePullError(err, driverConfig.ImageName)
+	}
+	d.logger.Printf("[DEBUG] driver.docker: docker pull %s:%s succeeded", repo, tag)
+	return nil
+}
 
-		// Now that we have the image we can get the image id
-		dockerImage, err = client.InspectImage(image)
-		if err != nil {
-			d.logger.Printf("[ERR] driver.docker: failed getting image id for %s: %s", image, err)
-			return nil, fmt.Errorf("Failed to determine image id for `%s`: %s", image, err)
-		}
+// loadImage creates an image by loading it from the file system
+func (d *DockerDriver) loadImage(driverConfig *DockerDriverConfig, client *docker.Client, taskDir string) error {
+	archive := filepath.Join(taskDir, allocdir.TaskLocal, driverConfig.LoadImage)
```

Comment:
I am not sure you want to do that because artifacts can be downloaded to arbitrary locations